pkg/translate/translate.go

```diff
@@ -115,10 +117,22 @@ type Translation struct {
 
 // NewTranslator creates a new Translator for minorVersion and returns a ptr to it.
 func NewTranslator(minorVersion version.MinorVersion) (*Translator, error) {
+	return newTranslator(minorVersion, maxFallbackNum)
+}
+
+func newTranslator(minorVersion version.MinorVersion, fallbackNum uint) (*Translator, error) {
 	v := fmt.Sprintf("%s.%d", minorVersion.MajorVersion, minorVersion.Minor)
 	f := "translateConfig/translateConfig-" + v + ".yaml"
 	b, err := vfs.ReadFile(f)
 	if err != nil {
+		if fallbackNum > 0 && minorVersion.Minor > 0 {
```

Comment:
It would be good to create a function with the fallback logic and a clear explanation of the version selection logic. This should test for existence of needed assets rather than trying to create a translator.

Reply:
The test for existence of assets is actually inside the translator creation function. A more readable way here is to use iteration instead of recursion. wdyt?